### PR TITLE
Support aarch64/arm64 by using dup3 instead of dup2

### PR DESCRIPTION
**Pull request by the-eater**

### CODE REVIEW

File: log/log_redirect_arm64.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package log
```
Comment (Freeaqingme) [inline log/log_redirect_arm64.go:1]: I think this should include a build flag as well to only build on arm64?

File: log/log_redirect_arm64.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	}
+
+	logFile, err := os.OpenFile(logPath, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0644)
+	if err != nil {
+		Log.Fatal(err)
+	}
+
+	syscall.Dup3(int(logFile.Fd()), 1, 0)
+	syscall.Dup3(int(logFile.Fd()), 2, 0)
+}
```
Comment (Freeaqingme) [inline log/log_redirect_arm64.go]: Missing new line

File: log/log_redirect.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	}
+
+	logFile, err := os.OpenFile(logPath, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0644)
+	if err != nil {
+		Log.Fatal(err)
+	}
+
+	syscall.Dup2(int(logFile.Fd()), 1)
+	syscall.Dup2(int(logFile.Fd()), 2)
+}
```
Comment (Freeaqingme) [inline log/log_redirect.go]: Missing new line

File: log/log_redirect_arm64.go (hunk 2 above)
Comment (Freeaqingme) [reply log/log_redirect_arm64.go]: There's still some code duplication that could be abstracted. Did you consider this?

**Review by Freeaqingme — COMMENTED** [review]
There's still some code duplication that could be abstracted. Did you consider this?

Other than the few comments, it looks good to me!

File: log/log_redirect_arm64.go (hunk 1 above)
Comment (the-eater) [reply log/log_redirect_arm64.go:1]: As quoted from https://golang.org/pkg/go/build/#hdr-Build_Constraints

> If a file's name, after stripping the extension and a possible _test suffix, matches any of the following patterns:
> 
> ```
> *_GOOS
> *_GOARCH
> *_GOOS_GOARCH
> ```
> 
> (example: source_windows_amd64.go) where GOOS and GOARCH represent any known operating system and architecture values respectively, then the file is considered to have an implicit build constraint requiring those terms (in addition to any explicit constraints in the file).

**Pull request merged**